nemo_skills/code_execution/math_grader.py

```diff
@@ -84,9 +84,33 @@
 import contextlib
 import re
 import signal
+from importlib.metadata import PackageNotFoundError, version
 from math import isclose
 from typing import Union
 
+from rich import print as rprint
+
+
+def _check_antlr_version():
+    "Function for checking the antlr package version."
+    # Check antlr version
+    PACKAGE_NAME = 'antlr4-python3-runtime'
+    REQUIRED_VERSION = '4.11.0'
+
+    try:
+        installed_version = version(PACKAGE_NAME)
+        if installed_version != REQUIRED_VERSION:
+            rprint(
+                f"[bold red]Package version mismatch: {installed_version} (required: {REQUIRED_VERSION})[/bold red]"
+            )
+            sys.exit(1)
+    except PackageNotFoundError:
+        rprint(f"[bold red]Package {PACKAGE_NAME} not found[/bold red]")
+        sys.exit(1)
```

Comment:
```suggestion
        raise
```
Better to reraise the exception after printing the instruction on what to do, since it will give people an exact traceback of where the error is coming from

Reply:
That's a good point. Replaced print + sys.exit with a raise command.